server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -927,22 +1005,27 @@ public Pool find(String poolId) {
                     return null;
                 }
 
-                // now make the entitlements
-                for (PoolQuantity entry : bestPools) {
-                    entitlements.add(addOrUpdateEntitlement(consumer, entry.getPool(),
-                        null, entry.getQuantity(), false, CallerType.BIND));
-                }
-
-                return entitlements;
+                return entitleByPools(consumer, convertToMap(bestPools));
             }
             catch (EntitlementRefusedException e) {
-                String key = e.getResult().getErrors().get(0).getResourceKey();
-
-                if (key.equals("rulefailed.no.entitlements.available") && retries-- > 0) {
-                    log.info(
-                        "Entitlements exhausted between select best pools and bind operations; retrying"
-                    );
-
+                // if there are any pools that had only one error, and that was
+                // an availability error, try again
+                boolean retry = false;
+                if (retries > 0) {
+                    for (String poolId : e.getResults().keySet()) {
+
+                        List<ValidationError> errors = e.getResults().get(poolId).getErrors();
+                        if (errors.size() == 1 && errors.get(0).getResourceKey()
+                                        .equals("rulefailed.no.entitlements.available")) {
+                            retry = true;
+                            break;
+                        }
+                    }
+                }
+                if (retry) {
+                    log.info("Entitlements exhausted between select best pools and bind operations;" +
+                        " retrying");
+                    retries--;
                     continue;
```

Comment:
It's in a while loop. The block starts way up on 983, which github is hiding by default.